libraries/mysql_service_manager_sysvinit.rb

```diff
@@ -39,7 +39,7 @@ class MysqlServiceManagerSysvinit < MysqlServiceBase
       end
 
       service mysql_name do
-        provider Chef::Provider::Service::Init::Redhat if node['platform_family'] == 'redhat'
+        provider Chef::Provider::Service::Init::Redhat if %w(redhat suse).include?(node['platform_family'])
```

Comment:
`rhel` is the platform_family, and it should be `%w(rhel fedora suse)`

i'm not sure why you can't rely on the provider resolver picking the correct service provider here and just delete these lines.

i guess conceivably you could force it to use this class on a systemd-supporting host and then wind up with the system provider when you really wanted to force the sysvinit provider -- but can we simply not give any fucks about that use case?